packages/fast-components-react-base/src/viewport-positioner/viewport-positioner.tsx

```diff
@@ -0,0 +1,1057 @@
+        viewport: void 0,
+        horizontalPositioningMode: void 0,
+        defaultHorizontalPosition: void 0,
+        horizontalThreshold: void 0,
+        horizontalAlwaysInView: void 0,
+        verticalPositioningMode: void 0,
+        defaultVerticalPosition: void 0,
+        verticalThreshold: void 0,
+        verticalAlwaysInView: void 0,
+        fixedAfterInitialPlacement: void 0,
+        disabled: void 0,
+    };
+
+    private rootElement: React.RefObject<HTMLDivElement> = React.createRef<
+        HTMLDivElement
+    >();
+
+    private openRequestAnimationFrame: number = null;
+
+    private collisionDetector: IntersectionObserver;
+    private resizeDetector: ResizeObserverClassDefinition;
+
+    private viewportRect: ClientRect | DOMRect;
+    private positionerRect: ClientRect | DOMRect;
+    private anchorTop: number = 0;
+    private anchorRight: number = 0;
+    private anchorBottom: number = 0;
+    private anchorLeft: number = 0;
+    private anchorHeight: number = 0;
+    private anchorWidth: number = 0;
+
+    private scrollTop: number = 0;
+    private scrollLeft: number = 0;
+
+    /**
+     * constructor
+     */
+    constructor(props: ViewportPositionerProps) {
+        super(props);
+
+        this.state = {
+            disabled: true,
+            noOberverMode: false,
+            xTransformOrigin: "left",
+            yTransformOrigin: "top",
+            xTranslate: 0,
+            yTranslate: 0,
+            top: null,
+            right: null,
+            bottom: null,
+            left: null,
+            currentHorizontalPosition:
+                ViewportPositionerHorizontalPositionLabel.uncontrolled,
+            currentVerticalPosition: ViewportPositionerVerticalPositionLabel.uncontrolled,
+            defaultHorizontalPosition: this.ConvertHorizontalPositionToLabel(
+                this.props.horizontalPositioningMode,
+                this.props.defaultHorizontalPosition
+            ),
+            defaultVerticalPosition: this.ConvertVerticalPositionToLabel(
+                this.props.verticalPositioningMode,
+                this.props.defaultVerticalPosition
+            ),
+            initialLayoutComplete: false,
+        };
+    }
+
+    public componentDidMount(): void {
+        this.checkComponentConfig();
+        this.requestFrame();
+    }
+
+    public componentWillUnmount(): void {
+        this.disableComponent();
+    }
+
+    public componentDidUpdate(prevProps: ViewportPositionerProps): void {
+        if (
+            prevProps.viewport !== this.props.viewport ||
+            prevProps.anchor !== this.props.anchor ||
+            prevProps.disabled !== this.props.disabled
+        ) {
+            this.checkComponentConfig();
+        }
+    }
+
+    /**
+     * Renders the component
+     */
+    public render(): React.ReactElement<HTMLDivElement> {
+        return (
+            <div
+                ref={this.rootElement}
+                {...this.unhandledProps()}
```

Comment:
This should occur after all other attributes or it could unintentionally override them

Reply:
yep